pkgs/servers/gemini/agate/default.nix

```diff
@@ -1,20 +1,23 @@
-{ lib, stdenv, fetchFromGitHub, rustPlatform, libiconv, Security }:
+{ lib, stdenv, nixosTests, fetchFromGitHub, rustPlatform, libiconv, Security }:
 
 rustPlatform.buildRustPackage rec {
   pname = "agate";
-  version = "3.1.0";
+  version = "3.1.3";
 
   src = fetchFromGitHub {
     owner = "mbrubeck";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-miIMz4Lk4R5So96Ceqe1Fl5ozpf47qWq0GgtKFDDCCA=";
+    sha256 = "sha256-gI03MJgqU+nTBJBNoj1kI/UOfXVQUcwZsRS9bK8ffU4=";
   };
 
-  cargoSha256 = "sha256-NZpqCl37dZUvXmn4Q1Pvbz3LSxk1s0s5x1CBO0POA/4=";
+  cargoSha256 = "sha256-hPGzTUe9WWUIZd7yuLNtD2wirB8EuhAeoXV7jWZzuQI=";
 
   buildInputs = lib.optionals stdenv.isDarwin [ libiconv Security ];
 
+  # port collision issues
```

Comment:
On darwin or only on Linux?

Reply:
It's due to the tests trying to use the same port twice at the same time so I imagine on both systems

It looks like 3.1.3 actually fixed that:

> the fix for dual stack listening from 3.1.2 was executed asynchronously and would thus
sometimes fail. starting the listeners on different socket addresses is now synchronous

https://github.com/mbrubeck/agate/releases/tag/v3.1.3